docs/data/routes/docs/nextjs/troubleshooting/en.md

```diff
@@ -0,0 +1,156 @@
+---
+name: troubleshooting
+routeTemplate: ./data/component-templates/article.yml
+title: Documentation
+---
+
+# Troubleshooting
+
+This section has advice for troubleshooting issues with JSS development
+
+## Server-side JavaScript errors
+
+If you encounter unexpected JavaScript errors during the `npm install` or `build` steps, especially errors that cannot be reproduced by other team members, check the versions of Node and npm being used. From terminal, run:
+```
+node -v
+npm -v
+```
+Note that JSS is tested using the Long Term Support (TLS) versions of Node. These are typically one major version behind the latest official Node version.
+
+Note that the Node/npm version used by your CI/Production environments may be different from what is used by your local environment. If a project is not pinned to a specific Node/npm version in `package.json`, it is common for deployment agents to build using either the newest version available or an environment-specific "default" version.
+
+If you need to switch Node version on your local environment for testing, 3rd party packages like [n](https://github.com/tj/n) or [nvm]() can help with this task.
+
+System Pre-requisites
+Running Node LTS version (use node -v to check)
+
+## App does not build or Home Page shows 404
+
+Use the GraphiQL interface to verify that the sitemap query returns the expected list of page routes.
+
+```
+# YOUR_PATH should be the ID of your site root (home page) in lower-case, with dashes removed
+# YOUR_LANGUAGE should be your default language, as defined in package.json
+
+query{
+  search(where: {
+    AND:[
+      {
+      name:"_path",
+      value:"YOUR_PATH"
+      },
+      {
+      name:"_language",
+      value:"YOUR_LANGUAGE"
+      },
+      {
+      name:"_hasLayout",
+      value :"true"
+      }
+    ]
+  })
+  { 
+    total
+    results {
+        url {
+          path
+        }
+      }
+  }
+}
+```
+
+If you do not see expected results, try the following steps:
+
+1. Populate Solr Schema (Sitecore Control Panel > “Populate Solr Managed Schema”)
+
+2. Rebuild indexes (Sitecore Control Panel > “Indexing Manager”)
+
+## Errors regarding SSL certificates
+
+Error examples
+> Error: unable to verify the first certificate
+>
+> Error: UNABLE_TO_VERIFY_LEAF_SIGNATURE
+
+> UnauthorizedError: invalid signature
+
+If you are working locally, make sure that you have [followed the steps](todo://add_link_to_new_page) for enabling self-signed certificates in Node. Check that you have the Node Extra Certificates path in your system variables.
+
+Node that the steps are different, depending on whether you're using a containerized or on-prem installation of Sitecore.
+
+## Errors when generating GraphQL introspection data
```

Comment:
Add a link to the introspection doc (/docs/nextjs/graphql/introspection) somewhere here to provide some context?

Reply:
Context added